src/Kiota.Builder/PathSegmenters/RubyPathSegmenter.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Text;
+using Kiota.Builder.Extensions;
+
+namespace Kiota.Builder {
+    public class RubyPathSegmenter : CommonPathSegmenter
+    {
+        public RubyPathSegmenter(string rootPath, string clientNamespaceName) : base(rootPath, clientNamespaceName) { }
+        public override string FileSuffix => ".rb";
+        public override string NormalizeFileName(string elementName) => ToSnakeCase(elementName);
+        public override string NormalizeNamespaceSegment(string segmentName) => ToSnakeCase(segmentName);
+
+        public string ToSnakeCase(string text)
```

Comment:
We have a stringExtensions class that already contains methods for casing management, move it there.